tests/v0x01/test_asynchronous/test_error_msg.py

```diff
@@ -14,8 +14,35 @@ def setUpClass(cls):
         """Setup TestStruct."""
         super().setUpClass()
         super().set_raw_dump_file('v0x01', 'ofpt_error_message')
-        super().set_raw_dump_object(ErrorMsg, xid=12,
-                                    error_type=ErrorType.OFPET_BAD_REQUEST,
-                                    code=BadRequestCode.OFPBRC_BAD_STAT,
-                                    data=BinaryData('object_test'))
+        super().set_raw_dump_object(ErrorMsg,
+                                    **TestErrorMessage.get_attributes())
         super().set_minimum_size(12)
+
+    def test_pack_error_msg(self):
+        """Test Pack a sample ErrorMessage."""
+        error_msg = ErrorMsg(**TestErrorMessage.get_attributes())
+        expected_data = b'\x01\x01\x00\x0c\x00\x00\x00\x0c\x00\x01\x00\x02'
+        self.assertEqual(error_msg.pack(), expected_data)
+
+    def test_unpack_error_msg(self):
+        """Test Unpack a sample ErrorMsg."""
+        packed_data = b'\x01\x01\x00\x1b\x00\x00\x00\x18\x00\x03\x00\x02FLOW'
+
+        attributes = TestErrorMessage.get_attributes(
+            xid=24, error_type=ErrorType.OFPET_FLOW_MOD_FAILED,
+            code=FlowModFailedCode.OFPFMFC_EPERM, data=b'FLOW')
+
+        unpacked = ErrorMsg(**attributes)
+
+        sample_unpacked = ErrorMsg(xid=24)
+        sample_unpacked.unpack(packed_data[8:])
```

Comment:
@cemsbr actually the header is not discarded on line 38. That slice (`[8:]`) is just because the unpack method does not handle the Header of the messages (by design for now). So, if you want to set the xid of a message, this can be done while instantiating the object (like in line 37) or after that by setting `msg.header.xid = <xid>`.